National context sectoral activity

aabdaab opened the pull request:
This PR adds sectoral activity to National Context.
We've got 3 cases in here
1. When indicator (API's sector) is `Primary source of emissions`
  * Year is selectable
  * We show all sectors on the map
  * We don't filter by activities
  * Values displayed in the tooltip are summarized per sector

2. When indicator is `Adaptation`
  * No filtering by year therefore year dropdown disappears and we don't show PlayTimeline component
  * Yes/No/No data in the legend

3. When indicator is other than `Primary source...` or `Adaptation`
  * Year is selectable
  * We color the map by activities (subsectors of sectors)

I had to add map component to the repo instead of using the one from cw-components for 2 reasons:
- there was an issue with linking custom tooltip
- on the way it turned out we'd need to add more specific data to for the map and I didn't want to customize this much the map in cw-components.

I also added rake task for updating translations that combines running two rake tasks that had to be ran up until now.
Run: `rake db:update_translations` for updating translations.
I also updated `cw-components` so you'll probably need to run `rm yarn.lock`, `rm -rf node_modules` && `yarn` again. Enjoy 🤓 

Screenshot:

![image](https://user-images.githubusercontent.com/6136899/49599563-42b02800-f979-11e8-96d7-d1bea9368a5a.png)

simaob commented:
Nice description and I'm glad I saw that I had to run that rake task :D as I wasn't seeing the link 🤦‍♂️

aabdaab commented:
Play button => Pause button requires changes to the timeline component in cw-components but I think it's worth it! didn't even realize that.
The jumping should be easy to fix; let me check

aabdaab commented:
Jumping fixed

simaob commented:
Alright =) smoother! =) Just noticed that we are probably missing the unit on the tooltips. I'm guessing that it's MtCO2e. But we might need to confirm or divid by 1k. (Check with Kamil the fixes he has been doing on SA)!

simaob commented:
Looks great now. =) Not sure if Kamil has any feedback, otherwise ready to be merged.

kfilip94 commented:
I'll check it now :)

kfilip94 commented on app/javascript/app/components/map/map-component.jsx:
```diff
@@ -0,0 +1,298 @@
+    } = this.props;
+
+    const getTooltip = name => {
+      const path = paths.find(p => p.properties.name === name);
```
I think we can use destructuring here like for whole return method below:
`const { tooltipValue, tooltipUnit, sector, selectedYear } = path.properties;`

kfilip94 commented on app/javascript/app/pages/national-context/sectoral-activity/sectoral-activity-selectors.js:
```diff
@@ -0,0 +1,418 @@
+    return { paths, legend };
+  };
+
+export const getPathsWithStyles = (provincesData, selectedOptions, sectors) => {
```
It's a really long selector, maybe you could extract blocks, especially `Object.keys...` to separates methods, it will be more readable when block like this will be named method

aabdaab commented:
^ Kamil, please take a look. I refactored the selectors, looks much better now. Also added `cacheId` to Geography for better performance. Changed key functions to selectors for better performance as well.

aabdaab commented:
Done! Thanks a lot for the review 💯
